tests/distributions/pareto_test.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+from jax import numpy as jnp
+from jax import random
+
+from mcx.distributions import Pareto
+
+
+@pytest.fixture
+def rng_key():
+    return random.PRNGKey(0)
+
+
+def pareto_mean(shape, scale):
+    if shape > 1:
+        return shape * scale / (shape - 1.0)
+    else:
+        return jnp.inf
+
+
+def pareto_variance(shape, scale):
+    if shape <= 2:
+        return jnp.inf
+    else:
+        numerator = (scale ** 2) * shape
+        denominator = ((shape - 1) ** 2) * (shape - 2)
+        return numerator / denominator
+
```

Comment:
Great addition! However, before I merge we'll need to add tests for the shape and the support! Would you mind adding those?

Reply:
indeed, was planning on it when I get some time, hopefully in the next few days!

Reply:
Is there anything I can do to help?

Reply:
I'll let you know when I get to it this weekend. Last 2.5 weeks I had a kidney stone which involved two surgeries and like 5 nights in hospitals, but things seem to be resolved now. 2021 has not been my year in terms of health. Nonetheless, I should finally have some time this weekend!

Reply:
ok I found some time to add more tests - but I'm having one issue with a failing test for the variance in the case when the shape parameter is <= 2 and I'm not entirely sure what I've implemented wrong. Not being totally familiar with the Pareto distribution i've kind of just followed the information on https://en.wikipedia.org/wiki/Pareto_distribution which is stating that the variance should be infinite when the shape parameter is <= 2, however this is not the case in the current implementation. I'd appreciate some feedback!

Reply:
Extensive test suite, great job!

Remember that we defined `shape = b` in this case. The variance should thus be  theoretically infinite when $shape < 1$ per the fomulae above.

Then, if you measure the variance of samples drawn from the distribution, you should get a very large number but not strictly $\infty$. You can check that $\sigma > 10  \mu$ for instance when $0 < shape < 1$. It would also be nice to check that $\mu \rightarrow \infty$  when $shape < 0$.

Reply:
Alright, I'll update the tests to reflect this. Thanks for the clarification!

Reply:
Sorry was away from this too long and am a bit confused because in your suggestion you are using $\sigma$ and $\mu$ notation, and I'm a bit confused as to what you are referring to, when you say "The variance should thus be theoretically infinite when $shape < 1$ per the fomulae above." I'm not sure what formulae you are exactly referring too because in the way I've implemented it, having a $shape < 1$ doesn't result in the variance being infinite:

```
        numerator = (scale ** 2) * shape
        denominator = ((shape - 1) ** 2) * (shape - 2)
        return numerator / denominator
```

I get that for that variance of the samples won't strictly be $\infty$, but I think I have implemented the Pareto distribution incorrectly but can't figure out what I've done wrong. Would need some additional assistance, thanks!